pkg/mongoproxy/plugins/schema/types.go

```diff
@@ -488,6 +489,20 @@ func (c *CollectionField) Validate(ctx context.Context, v interface{}, denyUnkno
 				}
 			}
 		}
+	case OBJECT_ARRAY:
+		switch vTyped := v.(type) {
+		case bson.A:
+			ok = true
+			for _, d := range vTyped {
+				dType, ok := d.(bson.D)
+				if !ok {
+					return fmt.Errorf("#{dType}: is not object")
+				}
+				if err := Validate(ctx, dType, c.SubFields, denyUnknownFields, isUpdate); err != nil {
+					return err
+				}
```

Comment:
can we use `err := c.ValidateElement(ctx, d, OBJECT)`?

Reply:
I don't think so, in ValidateElement we don't have OBJECT https://github.com/wish/mongoproxy/blob/master/pkg/mongoproxy/plugins/schema/types.go#L361